applications/TopologyOptimizationApplication/custom_utilities/topology_updating_utilities.h

```diff
@@ -0,0 +1,350 @@
+//                   Eric Gonzales
+//
+// ==============================================================================
+
+#if !defined(KRATOS_TOPOLOGY_UPDATING_UTILITIES_H_INCLUDED)
+#define  KRATOS_TOPOLOGY_UPDATING_UTILITIES_H_INCLUDED
+
+// System includes
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+// External includes
+#include <pybind11/pybind11.h>
+#include <boost/numeric/ublas/matrix.hpp>
+#include <boost/numeric/ublas/vector.hpp>
+#include <boost/numeric/ublas/io.hpp>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+#include "includes/model_part.h"
+#include "includes/process_info.h"
+
+
+// Application includes
+#include "topology_optimization_application.h"
+#include "structure_response_function_utilities.h"
+
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Solution utility that updates response values for next iteration.
+/** Detail class definition.
+
+ */
+
+class TopologyUpdatingUtilities
+{
+public:
+
+	///@name Type Definitions
+	///@{
+
+	/// Pointer definition of TopologyUpdatingUtilities
+	KRATOS_CLASS_POINTER_DEFINITION(TopologyUpdatingUtilities);
+
+	///@}
+	///@name Life Cycle
+	///@{
+
+	/// Default constructor.
+	TopologyUpdatingUtilities( ModelPart& model_part )
+	: mrModelPart(model_part)
+	{
+	}
+
+	/// Destructor.
+	virtual ~TopologyUpdatingUtilities()
+	{
+	}
+
+
+	///@}
+	///@name Operators
+	///@{
+
+
+	///@}
+	///@name Operations
+	///@{
+
+	// ---------------------------------------------------------------------------------------------------------------------------------------------
+	// --------------------------------- UPDATE DENSITIES  -----------------------------------------------------------------------------------------
+	// ---------------------------------------------------------------------------------------------------------------------------------------------
+
+	/// Finds the value of the X_PHYS (density) and updates it into the optimization problem
+	void UpdateDensitiesUsingOCMethod( char update_type[], double volfrac, double greyscale , double OptItr , double qmax)
+	{
+		KRATOS_TRY;
+
+		if ( strcmp( update_type , "oc_algorithm" ) == 0 ){
+			clock_t begin = clock();
+			std::cout << "  Optimality Criterion Method (OC) chosen to solve the optimization problem" << std::endl;
+
+			// Check if Grey Scale Filter should be used
+			double q = 1;
+			if (greyscale == 1)
+			{
+				if (OptItr < 10)
+					q = 1;
+				else
+					q = std::min(qmax, 1.1*q);
+
+				std::cout << "  Grey Scale Filter activated, q = " << q << std::endl;
+			}
+			else
+			{
+				std::cout << "  Grey Scale Filter deactivated, q = " << q << std::endl;
+			}
+
+			// Update Densities procedure
+			double l1     = 0.0;
+			double l2     = 1000000000000.0; ///10000000000.0
```

Comment:
Tabulations..., and use instead
```suggestion
			double l2     = 1.0e12; ///10000000000.0
```